js/search.js

```diff
@@ -962,11 +961,39 @@
 				if (id === 'hiddenpowergrass') isViable = (moves.indexOf('energyball') < 0 && moves.indexOf('gigadrain') < 0);
 				if (id === 'hiddenpowerice') isViable = (moves.indexOf('icebeam') < 0 && template.id !== 'xerneas');
 				if (id === 'icywind') isViable = (toId(set.species).substr(0, 6) === 'keldeo');
+				if (id === 'leechlife') isViable = (this.gen > 6);
 				if (id === 'reflecttype') isViable = (toId(set.species) in {latias:1, starmie:1});
 				if (id === 'selfdestruct') isViable = (this.gen < 5 && moves.indexOf('explosion') < 0);
 				if (id === 'skyattack') isViable = (toId(set.species) === 'hawlucha');
 				if (id === 'stunspore') isViable = (moves.indexOf('thunderwave') < 0);
-				if (id === 'toxic') isViable = (this.gen > 1);
+				if (this.gen === 1) {
+					// Usually viable for Gen 1
+					if (id === 'acidarmor' || id === 'amnesia' || id === 'barrier' || id === 'bind' || id === 'clamp' || id === 'confuseray' || id === 'counter' || id === 'firespin' || id === 'hyperbeam' || id === 'mirrormove' || id === 'pinmissile' || id === 'razorleaf' || id === 'sing' || id === 'slash' || id === 'sludge' || id === 'twineedle' || id === 'wrap') isViable = true;
+
+					// Usually not viable for Gen 1
+					if (id === 'disable' || id === 'firepunch' || id === 'icepunch' || id === 'jumpkick' || id === 'leechseed' || id === 'quickattack' || id === 'roar' || id === 'thunderpunch' || id === 'toxic' || id === 'triattack' || id === 'whirlwind') isViable = false;
+
+					// Viable only when certain moves aren't present
+					if (id === 'doubleedge') isViable = (moves.indexOf('bodyslam') < 0);
+					if (id === 'hypnosis') isViable = (moves.indexOf('sleeppowder') < 0);
+					if (id === 'megadrain') isViable = (moves.indexOf('razorleaf') < 0 && moves.indexOf('surf') < 0);
+					if (id === 'megakick') isViable = (moves.indexOf('hyperbeam') < 0);
+					if (id === 'reflect') isViable = (moves.indexOf('barrier') < 0 && moves.indexOf('acidarmor') < 0);
+
+					// Viable for specific types
+					if (id === 'substitute') isViable = (template.types[0] === 'Ground' || template.types[1] === 'Ground');
```

Comment:
It blocks Confuse Ray and Toxic and the secondary status from moves such as Flamethrower and Ice Beam, but doesn't block thunder wave, hypnosis, or any other sleep move. As such it's only really seen as a filler on Pokemon which can force switches that don't have much else to be doing (usually ground-types: Rhydon is the only common user in OU, whilst Dugtrio / Nidoking / Nidoqueen can use it in (PP's) lower tiers). The other cases where it's used, if you don't mind me talking in the context of very low tiers on PP, in RBY 5U [PU] on PP, Seadra [a top Pokemon, seen on every team] uses it frequently because of its ability to block the secondary effect of Ice Beam, and Pidgeot uses Substitute in combination with Toxic & Rest in RBY 6U [ZU] to wear down Kabutops.